code-challenges.test.js

```diff
@@ -13,47 +22,121 @@
-var num3 = -7
-// Expected output: "-7 is not divisible by three"
+describe("divByThree", () => {
+    var num1 = 15
+    // Expected output: "15 is divisible by three"
+    var num2 = 0
+    // Expected output: "0 is divisible by three"temperature", "database", "chopsticks", "mango", "deduction"
+    var num3 = -7
+    // Expected output: "-7 is not divisible by three"
+
+    it("decides if the number is evenly divisble by three or not", () => {
+        expect(divByThree(num1)).toEqual("15 is divisible by three")
+        expect(divByThree(num2)).toEqual("0 is divisible by three")
+        expect(divByThree(num3)).toEqual("-7 is not divisible by three")
+    })
+})
+// Result: Test failed, divByThree is not defined.
 
+// b) Create the function that makes the test pass.
 
+//Pseudo code: 
+// Create a function diveByThree
+// this function take a number as an input argument 
+// verify if a number is divByThree 
 
-// b) Create the function that makes the test pass.
+
+// const divByThree = (number) => {
+//     if (number % 3 === 0) {
+//         return `${number} is divisible by three`
+
+//     } else if (number % 3 !== 0) {
+//         return `${number} is not divisible by three`
+//     }
+
+//Result: Test passed; divByThree defined. 
 
 
 
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-var randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
-// Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
-var randomNouns2 = ["temperature", "database", "chopsticks", "mango", "deduction"]
-// Expected output: ["Temperature", "Database", "Chopsticks", "Mango", "Deduction"]
+describe("randomNouns", () => {
+    var randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+    // Expect output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+    var randomNouns2 = ["temperature", "database", "chopsticks", "mango", "deduction"]
+    // Expect output: ["Temperature", "Database", "Chopsticks", "Mango", "Deduction"]
 
 
+    it("decide function that takes in an array of words and returns an array with all the words capitalized", () => {
+        expect(randomNouns1.upperCase("Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew")
+        expect(randomNouns2.upperCase("Temperature", "Database", "Chopsticks", "Mango", "Deduction")
 
-// b) Create the function that makes the test pass.
+    })
+    // 
+    // Test had a faulty failure due to the return in line 103 which is blank. Not sure what is in that line. 
 
+    // b) Create the function that makes the test pass.
 
 
-// --------------------3) Create a function that takes in a string and logs the index of the first vowel.
+    const randomNouns(array) = => {
+        let randomNouns1 = ("streetlamp", "potato", "teeth", "conclusion", "nepheww");
```

Comment:
Test variables are used as the input to your function, it shouldn't be in your function itself. Again, your first code challenge was GREAT! Model your other challenges against the ones that are working.